discovery_engine_core/providers/src/filter.rs

```diff
@@ -35,182 +30,39 @@ impl Filter {
         self
     }
 
-    /// Add a market to filter with. All market are in "or" with each other.
-    pub fn add_market(mut self, market: Market) -> Self {
-        self.markets.push(market);
-
-        self
-    }
-
-    /// Add a [`SiteType`] to filter with. All site types are in "or" with each other.
-    pub fn add_site_type(mut self, site_type: SiteType) -> Self {
-        self.site_types.push(site_type);
-
-        self
-    }
-
     /// Build the expression.
     pub(crate) fn build(&self) -> String {
         let keywords = Expr::or_from_iter(self.keywords.iter().map(|k| format!("\"{}\"", k)));
-        let markets = Expr::or_from_iter(self.markets.iter());
-        let site_types = Expr::or_from_iter(self.site_types.iter());
-
-        keywords.and(markets).and(site_types).build()
+        keywords.build()
     }
 }
 
 /// Define area and language of interests.
 #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Market {
+pub(crate) struct Market {
```

Comment:
If we are going to keep then why not using it in the QueryParam instead of two distinct strings?

Reply:
Mmh, I'm actually not quite sure what happened here. I wanted to delete this struct as well, either me or IntelliJ made a mistake with the deletion refactor :D